### PR TITLE
Add bin tasks to streamline release process

### PR DESCRIPTION
**Pull request by ABaldwinHunter**
@codeclimate/review

### CODE REVIEW

**Comment by GordonDiggs** [comment]
LGTM

**Comment by wfleming** [comment]
The kind of wonky `prep-release` & `release` scripts are what they are on CLI because we have protected branches on that repo. We don't have protected branches on this repo, so this flow seems like overkill to me. I think it's actually making the release process more complicated. You should be able to just edit the `version.rb` file & run `rake release`?

**Comment by GordonDiggs** [comment]
I actually vote we turn on protected branches here

File: bin/prep-release [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+git add VERSION Gemfile.lock
+git commit -m "Release v$version"
+git push origin "$branch"
+
+branch_head=$(git rev-parse --short $branch)
+if command -v hub > /dev/null 2>&1; then
+  hub pull-request -F - <<EOF
+Release v$version
+
+https://github.com/codeclimate/codeclimate/compare/v$old_version...$branch_head
```
Comment (wfleming) [inline bin/prep-release]: Wrong repo URL here.

**Comment by ABaldwinHunter** [comment]
I'm okay with turning on branch protection here.

File: bin/prep-release (hunk 1 above)
Comment (ABaldwinHunter) [inline bin/prep-release]: ah good catch, thanks!

**Comment by ABaldwinHunter** [comment]
My other thought is that I slightly like a `prep-release` script. Even though we typically merge anyway, I like semver feedback sometimes.

**Comment by ABaldwinHunter** [comment]
Okay looks like there are some arguments for and against this change, so going to table for now.

**Comment by dblandin** [comment]
@ABaldwinHunter Is this PR still active?

**Comment by dblandin** [comment]
Going to close for now.

**Comment by ABaldwinHunter** [comment]
👍 thanks

**Pull request closed without merging**